src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5874,6 +5886,63 @@ private ScanTypeFlags ScanTupleType(out SyntaxToken lastTokenOfType)
             return ScanTypeFlags.NotType;
         }
 
+        private ScanTypeFlags ScanFunctionPointerType(out SyntaxToken lastTokenOfType)
+        {
+            Debug.Assert(IsFunctionPointerStart());
+            _ = EatContextualToken(SyntaxKind.DelegateKeyword);
+            _ = EatToken(SyntaxKind.AsteriskToken);
+
+            switch (CurrentToken.ContextualKind)
+            {
+                case SyntaxKind.CdeclKeyword:
+                case SyntaxKind.ManagedKeyword:
+                case SyntaxKind.StdcallKeyword:
+                case SyntaxKind.ThiscallKeyword:
+                case SyntaxKind.UnmanagedKeyword:
+                    _ = EatContextualToken(CurrentToken.ContextualKind);
+                    break;
+
+                default:
+                    if (CurrentToken.Kind != SyntaxKind.LessThanToken)
+                    {
+                        _ = EatToken();
+                    }
```

Comment:
what's going on here.  why do you just consume wahtever token is next?  For example, say the user had `delegate*;`  or `delegate*>` (where the `>` is the terminator of an ourter list). You'd just eat that token unilaterally?

I think this is trying to consume other identifiers in case the user mispells  the calling convention token. In that case, i would specifically check that this is an identifier (or maybe identifier/keyword) before consuming blindly.